spec/ruby/core/module/attr_spec.rb

```diff
@@ -136,14 +138,16 @@ def initialize
     -> { Class.new { attr o } }.should raise_error(TypeError)
   end
 
-  it "with a boolean argument emits a warning when $VERBOSE is true" do
-    -> {
-      Class.new { attr :foo, true }
-    }.should complain(/boolean argument is obsoleted/, verbose: true)
-  end
+  ruby_version_is ""..."3.1" do
+    it "with a boolean argument emits a warning when $VERBOSE is true" do
+      -> {
+        Class.new { attr :foo, true }
+      }.should complain(/boolean argument is obsoleted/, verbose: true)
+    end
 
-  it "is a public method" do
-    Module.should have_public_instance_method(:attr, false)
+    it "is a public method" do
+      Module.should have_public_instance_method(:attr, false)
+    end
```

Comment:
This should not be inside the version guard

Reply:
@eregon 
Thanks for your comments!
I've done it(https://github.com/ruby/ruby/pull/3386/commits/87dba5ef95519b9302b934b958a87db2751ecc9c)